src/Utils/intersectionOfArrays.ts

```diff
@@ -0,0 +1,13 @@
+export function intersectionOfArrays<T>(a: T[], b: T[]): T[] {
+    const output: T[] = [];
+    const inA: Map<T, boolean> = new Map<T, boolean>();
```

Comment:
Could we just use a `Set` here?

Reply:
We could, my thinking was however that looking up a key would be quicker than finding a value in a Set.

Reply:
Set.has() is O(1), no?

Reply:
I can't find any solid resources, but it seems to be `O(n)`

Reply:
Happy to change it to `Set` if you'd rather.

Reply:
Let's use set. Checking whether an item exists should be O(1) or otherwise be fast.